### PR TITLE
 Fixed wrong intro scene`s deletion objects, crazy moving of main cam…

### PR DESCRIPTION
**Pull request by tomluchowski**
…era, crazy moving of minimap camera

### CODE REVIEW

**Comment by oyvindln** [comment]
I'm a bit busy the next few days but I'll have a look at it when I have time.

File: source/gamemap/MiniMapCamera.cpp [hunk 1]
```diff
@@ -174,7 +174,9 @@ void MiniMapCamera::updateMinimapCamera()
     mMiniMapCamNode->setPosition(mCurCamPosX, mCurCamPosY, CAM_HEIGHT);
     mMiniMapCamNode->lookAt(Ogre::Vector3(mCurCamPosX, mCurCamPosY, 0.0),
                      Ogre::Node::TransformSpace::TS_WORLD);
-    mMiniMapCamNode->roll(orientation.getRoll());
+    Ogre::Quaternion qq;
+    qq.FromAngleAxis( orientation.getRoll() ,Ogre::Vector3::UNIT_Z);
```
Comment (akien-mga) [inline source/gamemap/MiniMapCamera.cpp:178]: Style wise it should be `qq.FromAngleAxis(orientation.getRoll(), Ogre::Vector3::UNIT_Z);`

File: source/render/RenderManager.cpp [hunk 2]
```diff
@@ -300,8 +300,9 @@ void RenderManager::removeEntityMenu(Ogre::Entity* ent)
     Ogre::SceneNode* entNode = mSceneManager->getSceneNode(ent->getName() + "_node");
     entNode->detachObject(ent);
     mMainMenuSceneNode->removeChild(entNode);
+    mSceneManager->destroySceneNode(entNode->getName() );
```
Comment (akien-mga) [inline source/render/RenderManager.cpp:303]: Style wise, the space before closing `)` is unnecessary and should be amended.

File: source/render/RenderManager.cpp [hunk 3]
```diff
@@ -300,8 +300,9 @@ void RenderManager::removeEntityMenu(Ogre::Entity* ent)
     Ogre::SceneNode* entNode = mSceneManager->getSceneNode(ent->getName() + "_node");
     entNode->detachObject(ent);
     mMainMenuSceneNode->removeChild(entNode);
+    mSceneManager->destroySceneNode(entNode->getName() );    
     mSceneManager->destroyEntity(ent);
-    mSceneManager->destroySceneNode(entNode->getName());
+
```
Comment (akien-mga) [inline source/render/RenderManager.cpp:305]: You can remove that empty line.

**Comment by akien-mga** [comment]
First commit seems fine to me, just some style changes needed (please amend the commit (see `git rebase HEAD~2` and `edit` first commit) to fix those).

Could you explain a bit what the meshes update does? Since it's all binary it's a bit hard to tell.

I'll pull and test the changes locally tomorrow.

**Comment by tomluchowski** [comment]
The binary update is ... binary update, binary representation of ogre3d meshes tend to change with each major release and  a special tool OgreMeshUpgrader always upgrades the meshes to the newest format, that way the game ( or engine if you wish ) won't print a message : " warning the mesh xyz is in older format, you should upgrade it as soon as possible" anymore.

**Comment by akien-mga** [comment]
What version did you upgrade them to though?

**Comment by akien-mga** [comment]
Just for the reference, many of the models are getting quite bigger. I confirm that the models before this PR trigger obsolete warnings though: `Warning: Troll.mesh is an older format ([MeshSerializer_v1.8]); you should upgrade it as soon as possible using the OgreMeshUpgrade tool.`

```
 models/Adventurer.mesh           | Bin 216279 -> 546672 bytes
 models/AdventurerBed.mesh        | Bin 2311 -> 8173 bytes
 models/Anvil.mesh                | Bin 108939 -> 205200 bytes
 models/Arena.mesh                | Bin 678 -> 680 bytes
 models/ArrowProjectile.mesh      | Bin 6131 -> 11117 bytes
 models/BasicAxe.mesh             | Bin 21729 -> 39523 bytes
 models/BasicHammer.mesh          | Bin 43995 -> 77875 bytes
 models/BasicShield.mesh          | Bin 47727 -> 85915 bytes
 models/BasicSword.mesh           | Bin 27529 -> 50149 bytes
 models/Bed.mesh                  | Bin 10931 -> 28965 bytes
 models/Bookcase.mesh             | Bin 115353 -> 257026 bytes
 models/Bookshelf.mesh            | Bin 28583 -> 52571 bytes
 models/Boulder.mesh              | Bin 18646 -> 40994 bytes
 models/Bow.mesh                  | Bin 16811 -> 53895 bytes
 models/Cannon.mesh               | Bin 13732 -> 49157 bytes
 models/Cannonball.mesh           | Bin 6406 -> 13294 bytes
 models/Casino.mesh               | Bin 409 -> 678 bytes
 models/CasinoPokerTable.mesh     | Bin 32197 -> 59605 bytes
 models/CasinoWallBeer.mesh       | Bin 71055 -> 132744 bytes
 models/CaveHornet.mesh           | Bin 173303 -> 501419 bytes
 models/CelticCross.mesh          | Bin 9219 -> 17133 bytes
 models/Chicken.mesh              | Bin 24542 -> 61555 bytes
 models/ChickenCoop.mesh          | Bin 24255 -> 48906 bytes
 models/Chimney.mesh              | Bin 46309 -> 88257 bytes
 models/Claimed_fl_0000.mesh      | Bin 11023 -> 20184 bytes
 models/Claimed_fl_0100.mesh      | Bin 7351 -> 13434 bytes
 models/Claimed_fl_0101.mesh      | Bin 6127 -> 11184 bytes
 models/Claimed_fl_0110.mesh      | Bin 4903 -> 8934 bytes
 models/Claimed_fl_1110.mesh      | Bin 3679 -> 6684 bytes
 models/Claimed_fl_1111.mesh      | Bin 1231 -> 2184 bytes
 models/Claimed_gd_1111.mesh      | Bin 319 -> 588 bytes
 models/Cre_vision_indicator.mesh | Bin 809 -> 1883 bytes
 models/Crossbow.mesh             | Bin 28989 -> 78693 bytes
 models/Crypt.mesh                | Bin 411 -> 680 bytes
 models/Cultist.mesh              | Bin 165970 -> 414584 bytes
 models/DarkElf.mesh              | Bin 846796 -> 2182792 bytes
 models/Defender.mesh             | Bin 758941 -> 2130624 bytes
 models/Dirt0.mesh                | Bin 316 -> 585 bytes
 models/Dirt100.mesh              | Bin 316 -> 585 bytes
 models/Dirt101.mesh              | Bin 456 -> 921 bytes
 models/Dirt102.mesh              | Bin 596 -> 1257 bytes
 models/Dirt103.mesh              | Bin 1932 -> 4253 bytes
 models/Dirt104.mesh              | Bin 876 -> 1829 bytes
 models/Dirt105.mesh              | Bin 876 -> 1829 bytes
 models/Dirt_fl_0000.mesh         | Bin 12978 -> 51665 bytes
 models/Dirt_fl_0100.mesh         | Bin 11066 -> 44114 bytes
 models/Dirt_fl_0101.mesh         | Bin 9040 -> 36155 bytes
 models/Dirt_fl_0110.mesh         | Bin 7819 -> 24080 bytes
 models/Dirt_fl_1011.mesh         | Bin 10453 -> 31145 bytes
 models/Dirt_fl_1111.mesh         | Bin 6961 -> 19666 bytes
 models/Dirt_gd_1111.mesh         | Bin 316 -> 585 bytes
 models/Dojo.mesh                 | Bin 409 -> 678 bytes
 models/Dormitory.mesh            | Bin 419 -> 688 bytes
 models/Dragon.mesh               | Bin 30132 -> 77547 bytes
 models/DragonBed.mesh            | Bin 41789 -> 106163 bytes
 models/DungeonTemple.mesh        | Bin 320 -> 589 bytes
 models/DungeonTempleObject.mesh  | Bin 37671 -> 154194 bytes
 models/Dwarf1.mesh               | Bin 19960 -> 58312 bytes
 models/Dwarf2.mesh               | Bin 23043 -> 66897 bytes
 models/Elf.mesh                  | Bin 817178 -> 2101324 bytes
 models/Farm.mesh                 | Bin 409 -> 678 bytes
 models/FlyingSkull.mesh          | Bin 354165 -> 354167 bytes
 models/Forge.mesh                | Bin 411 -> 680 bytes
 models/Foundry.mesh              | Bin 364231 -> 670286 bytes
 models/Gnome.mesh                | Bin 23616 -> 66027 bytes
 models/Goblin.mesh               | Bin 470007 -> 1228314 bytes
 models/GoblinBed.mesh            | Bin 6887 -> 24919 bytes
 models/GoldstackLv1.mesh         | Bin 98845 -> 246691 bytes
 models/GoldstackLv2.mesh         | Bin 72889 -> 171433 bytes
 models/GoldstackLv3.mesh         | Bin 248413 -> 598723 bytes
 models/GoldstackLv4.mesh         | Bin 326077 -> 779671 bytes
 models/Grimoire.mesh             | Bin 74173 -> 121933 bytes
 models/Grindstone.mesh           | Bin 357085 -> 664310 bytes
 models/Hammock.mesh              | Bin 18979 -> 34859 bytes
 models/ImpBed.mesh               | Bin 15209 -> 27227 bytes
 models/Keeperhand.mesh           | Bin 75863 -> 192020 bytes
 models/Knight.mesh               | Bin 24943 -> 61769 bytes
 models/KnightCoffin.mesh         | Bin 238567 -> 693568 bytes
 models/KnightStatue.mesh         | Bin 240593 -> 649021 bytes
 models/KnightStatue2.mesh        | Bin 284097 -> 727812 bytes
 models/Kobold.mesh               | Bin 25171 -> 71148 bytes
 models/Kreatur.mesh              | Bin 39866 -> 124730 bytes
 models/Lamp.mesh                 | Bin 23257 -> 43128 bytes
 models/Lava0.mesh                | Bin 1016 -> 2090 bytes
 models/LavaSpawn.mesh            | Bin 439840 -> 439842 bytes
 models/Lava_gd_0000.mesh         | Bin 4132 -> 13981 bytes
 models/Lava_gd_0001.mesh         | Bin 3586 -> 11823 bytes
 models/Lava_gd_0010.mesh         | Bin 3478 -> 11519 bytes
 models/Lava_gd_0011.mesh         | Bin 2710 -> 8674 bytes
 models/Lava_gd_0100.mesh         | Bin 3586 -> 11823 bytes
 models/Lava_gd_0101.mesh         | Bin 3100 -> 9896 bytes
 models/Lava_gd_0110.mesh         | Bin 2710 -> 8674 bytes
 models/Lava_gd_0111.mesh         | Bin 1942 -> 5829 bytes
 models/Lava_gd_1000.mesh         | Bin 3478 -> 11519 bytes
 models/Lava_gd_1001.mesh         | Bin 2710 -> 8674 bytes
 models/Lava_gd_1010.mesh         | Bin 2992 -> 9592 bytes
 models/Lava_gd_1011.mesh         | Bin 1942 -> 5829 bytes
 models/Lava_gd_1100.mesh         | Bin 2710 -> 8674 bytes
 models/Lava_gd_1101.mesh         | Bin 1942 -> 5829 bytes
 models/Lava_gd_1110.mesh         | Bin 1942 -> 5829 bytes
 models/Lava_gd_1111.mesh         | Bin 652 -> 1142 bytes
 models/Library.mesh              | Bin 415 -> 684 bytes
 models/Lizardman.mesh            | Bin 182590 -> 584362 bytes
 models/LizardmanBed.mesh         | Bin 12917 -> 24851 bytes
 models/Longsword.mesh            | Bin 1865 -> 5237 bytes
 models/Monk.mesh                 | Bin 530656 -> 530658 bytes
 models/MysteryBox.mesh           | Bin 33171 -> 52437 bytes
 models/NatureMonster.mesh        | Bin 543005 -> 543007 bytes
 models/Orc.mesh                  | Bin 153458 -> 410710 bytes
 models/OrcBed.mesh               | Bin 21653 -> 35376 bytes
 models/PitDemon.mesh             | Bin 47268 -> 148206 bytes
 models/Podium.mesh               | Bin 19253 -> 35487 bytes
 models/Portal.mesh               | Bin 320 -> 589 bytes
 models/PortalObject.mesh         | Bin 148651 -> 337549 bytes
 models/PrisonGround.mesh         | Bin 692 -> 694 bytes
 models/RangerBed.mesh            | Bin 26819 -> 84101 bytes
 models/Rat.mesh                  | Bin 596260 -> 596262 bytes
 models/Roach.mesh                | Bin 170428 -> 429168 bytes
 models/Roulette.mesh             | Bin 141553 -> 141555 bytes
 models/Roundshield.mesh          | Bin 3269 -> 13577 bytes
 models/RunelordDwarf.mesh        | Bin 730980 -> 1801555 bytes
 models/Sabre.mesh                | Bin 1437 -> 7299 bytes
 models/Scarab.mesh               | Bin 37247 -> 90305 bytes
 models/Skull.mesh                | Bin 67097 -> 67099 bytes
 models/Slime.mesh                | Bin 4468 -> 11356 bytes
 models/SmallSpider.mesh          | Bin 29142 -> 76386 bytes
 models/Spider.mesh               | Bin 121551 -> 342855 bytes
 models/SpiderBed.mesh            | Bin 7331 -> 13624 bytes
 models/Spiketrap.mesh            | Bin 58028 -> 96433 bytes
 models/SquareSelector.mesh       | Bin 1505 -> 4589 bytes
 models/Staff.mesh                | Bin 2722 -> 7900 bytes
 models/StoneBridge.mesh          | Bin 17619 -> 32573 bytes
 models/StoneCoffin.mesh          | Bin 11619 -> 34581 bytes
 models/T2HammerGood.mesh         | Bin 75509 -> 141911 bytes
 models/T2ShieldEvil.mesh         | Bin 87845 -> 208399 bytes
 models/T2ShieldGood.mesh         | Bin 26381 -> 83883 bytes
 models/T2SwordGood.mesh          | Bin 44091 -> 124167 bytes
 models/T3HammerGood.mesh         | Bin 54717 -> 94078 bytes
 models/T3ShieldEvil.mesh         | Bin 9725 -> 33029 bytes
 models/T3ShieldGood.mesh         | Bin 18509 -> 38509 bytes
 models/T3SwordGood.mesh          | Bin 26913 -> 83655 bytes
 models/T4HammerGood.mesh         | Bin 124505 -> 197835 bytes
 models/T4ShieldEvil.mesh         | Bin 99221 -> 156624 bytes
 models/T4ShieldGood.mesh         | Bin 123071 -> 195906 bytes
 models/T4SwordGood.mesh          | Bin 37155 -> 113301 bytes
 models/TentacleAlbine.mesh       | Bin 36466 -> 120646 bytes
 models/TentacleBed.mesh          | Bin 55695 -> 95460 bytes
 models/TentacleGreen.mesh        | Bin 36465 -> 120645 bytes
 models/TortureGround.mesh        | Bin 678 -> 680 bytes
 models/TortureObject.mesh        | Bin 10087 -> 17133 bytes
 models/TrainingDummy1.mesh       | Bin 30795 -> 78435 bytes
 models/TrainingDummy2.mesh       | Bin 57323 -> 93601 bytes
 models/TrainingDummy3.mesh       | Bin 51183 -> 90410 bytes
 models/TrainingDummy4.mesh       | Bin 76175 -> 126521 bytes
 models/Treasury.mesh             | Bin 417 -> 686 bytes
 models/Troll.mesh                | Bin 248596 -> 537058 bytes
 models/TrollBed.mesh             | Bin 57111 -> 152443 bytes
 models/TrollRock.mesh            | Bin 39367 -> 148855 bytes
 models/WarBanner.mesh            | Bin 73570 -> 150984 bytes
 models/Water0.mesh               | Bin 562 -> 1074 bytes
 models/Water_gd_0000.mesh        | Bin 4133 -> 13982 bytes
 models/Water_gd_0001.mesh        | Bin 3587 -> 11824 bytes
 models/Water_gd_0010.mesh        | Bin 3479 -> 11520 bytes
 models/Water_gd_0011.mesh        | Bin 2711 -> 8675 bytes
 models/Water_gd_0100.mesh        | Bin 3587 -> 11824 bytes
 models/Water_gd_0101.mesh        | Bin 3101 -> 9897 bytes
 models/Water_gd_0110.mesh        | Bin 2711 -> 8675 bytes
 models/Water_gd_0111.mesh        | Bin 1943 -> 5830 bytes
 models/Water_gd_1000.mesh        | Bin 3479 -> 11520 bytes
 models/Water_gd_1001.mesh        | Bin 2711 -> 8675 bytes
 models/Water_gd_1010.mesh        | Bin 2993 -> 9593 bytes
 models/Water_gd_1011.mesh        | Bin 1943 -> 5830 bytes
 models/Water_gd_1100.mesh        | Bin 2711 -> 8675 bytes
 models/Water_gd_1101.mesh        | Bin 1943 -> 5830 bytes
 models/Water_gd_1110.mesh        | Bin 1943 -> 5830 bytes
 models/Water_gd_1111.mesh        | Bin 653 -> 1143 bytes
 models/WeaponShield1.mesh        | Bin 95695 -> 174201 bytes
 models/WeaponShield2.mesh        | Bin 100579 -> 216693 bytes
 models/Wizard.mesh               | Bin 37859 -> 102529 bytes
 models/WoodBridge.mesh           | Bin 8785 -> 16615 bytes
 models/WoodenDoor.mesh           | Bin 86307 -> 137635 bytes
 models/Workshop.mesh             | Bin 417 -> 686 bytes
 models/WorkshopMachine1.mesh     | Bin 64321 -> 121937 bytes
 models/WorkshopMachine2.mesh     | Bin 102025 -> 168941 bytes
 models/Wyvern.mesh               | Bin 34950 -> 95896 bytes
 models/lich.mesh                 | Bin 213238 -> 213240 bytes
 models/skeleton.mesh             | Bin 207763 -> 207765 bytes
```

**Comment by akien-mga** [comment]
I built the PR successfully locally, but I can't test it as OD crashes for me currently (when built against Ogre 1.10.12). That crash seems unrelated to this PR though.

**Comment by Bertram25** [comment]
Just in case, I tried upgrading the models once and this broke them or their texturing. You might want to run the special test level with all the monsters to make sure everything is ok first. Cheers, :)

**Comment by tomluchowski** [comment]
Many thanks Bertram, I will check it asap.
@akien-mga  : make sure you are not running the full drawn minimap, it is reported to cause game crash on larger maps.

**Comment by akien-mga** [comment]
> make sure you are not running the full drawn minimap, it is reported to cause game crash on larger maps.

That was it, thanks! I'll open an issue about it.

**Comment by akien-mga** [comment]
I tried the PR again with a different minimap to avoid the crash, and it seems to work fine. I did reproduce stutter in the camera movement before this PR, and the PR fixes it (Ogre 1.10). Not sure about what the minimap issue was, but I trust that it's a correct change.

I tested Model's fair and all models seem to be the same as they used to be. Yet when trying to upgrade one of them myself I get errors, e.g. on `Dragon.mesh` with OgreMeshUpgrader from 1.10.12:
```
Skeleton: Loading Dragon.skeleton
Error: Unable to load skeleton 'Dragon.skeleton' for Mesh 'conversion'. This Mesh will not be animated. You can ignore this message if you are using an offline tool.
```
So I think we should take this opportunity to review our meshes and make sure that our upgrade workflow is clear (and documented) before potentially losing data in the conversion process.

In the future, please put such unrelated changes in different PRs so that they can be reviewed independently.

**Comment by akien-mga** [comment]
I merged 66a452c manually as da9beafe with style fixes, thanks!

For the meshes upgrade, I prefer that we take our time to review the process before merging it, so it should be handled in a new PR dedicated to this change.

**Comment by Bertram25** [comment]
Good point. @tomluchowski How did you upgrade the mesh format? (Step by step)
I can have a look at that but it will have to wait for at least one more week.

**Comment by tomluchowski** [comment]
@Bertram25  @akien-mga he he : 

` for f in *.mesh; do OgreMeshUpgrader $f; done`

OgreMeshUpgrader which comes in ogre-tools for ogre1.10.11 ...

I might have not noticed all the errors by that command ...

**Pull request closed without merging**